myhpom/urls.py

```diff
@@ -29,6 +29,8 @@
         name='verify_account'),
 
     url(r'^download/(?P<path>.*)$', myhpom_views.irods_download, name='irods_download'),
+    url(r'^document/(?P<key>[A-Za-z0-9]+)/(?P<filename>[^/]+)/?$', 
```

Comment:
I've gone back and forth in my head on whether it makes sense to include the filename - its not strictly necessary...but I think...I like that you included it as its probably helpful for debugging. I guess an argument could be made that its information bleed b/c one can see the document's name in the URL, even if they don't have access to the doc itself b/c its expired. Does that seem like the kind of thing we should be concerned about (I could very well see users of the site putting their name into their document's filename, and I could see these URLs in particular sitting around on CF's own logging systems for a while...)?

Reply:
I would be okay omitting it. I went back and forth, too. The argument in favor of it in my head was that it provides a little validation. But frankly, the hash is long enough that I don't think we need it, and your point about information bleed is good. However, it's worth checking on how long CloudFactory keeps the pdfs around – because there's lots of private information there, and we want to make sure there's no bleed at that level, either!

Reply:
Good point - we forgot to ask about that...I was a little unclear what/ifs of caching they might do.